### PR TITLE
Dynamic music

### PR DESCRIPTION
**Pull request by LiamMillerSE**
Added a dynamic music system to the game

### CODE REVIEW

File: Assets/Audio/Music/DynamicAudioTest/TruckBossIntro.wav.meta [hunk 1]
```diff
@@ -0,0 +1,22 @@
+fileFormatVersion: 2
+guid: 825f538445a287540b25e0e7237942ff
```
Comment (toryjleo) [inline Assets/Audio/Music/DynamicAudioTest/TruckBossIntro.wav.meta:2]: Please change .wav files to .ogg and follow the formatting in ImportInfo.txt

File: Assets/Scripts/AI/DLevel.cs [hunk 2]
```diff
@@ -34,4 +34,12 @@ private void XTimer_Elapsed(object sender, ElapsedEventArgs e)
     {
         dangerLevel++;
     }
+
+    /// <summary>
```
Comment (toryjleo) [inline Assets/Scripts/AI/DLevel.cs:38]: Love it! Make us clean boys!

File: Assets/Scripts/Audio/Jukebox.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            int dl = DLevel.Instance.GetDangerLevel();
+            int variationIndex = 0;
+            for (; variationIndex < track.variations.Count; variationIndex++)
+            {
+                if (track.variations[variationIndex].dangerLevelStart >= dl)
+                {
+                    break;
+                }
+            }
+            Debug.unityLogger.Log("Danger level: " + dl + ". Queueing variation: " + variationIndex);
```
Comment (toryjleo) [inline Assets/Scripts/Audio/Jukebox.cs:72]: Didn't know this functionality existed

**Review by toryjleo — COMMENTED** [review]
Please change .wav files to .ogg and follow the formatting in ImportInfo.txt

File: Assets/Audio/Music/DynamicAudioTest/TruckBossIntro.wav.meta (hunk 1 above)
Comment (LiamMillerSE) [reply Assets/Audio/Music/DynamicAudioTest/TruckBossIntro.wav.meta]: Updated to follow standards

**Review by Tomerd-afk — APPROVED** [review]
:shipit:

**Pull request merged**